primitives/src/targeting/eval.rs

```diff
@@ -411,6 +603,94 @@ fn eval(input: &Input, output: &mut Output, rule: &Rule) -> Result<Option<Value>
 
             Some(Value::Bool(a.iter().any(|x| b.contains(x))))
         }
+        Function::In(first_rule, second_rule) => {
+            let a = eval(input, output, first_rule)?.ok_or(Error::TypeError)?;
+            let b = eval(input, output, second_rule)?
+                .ok_or(Error::TypeError)?
+                .try_array()?;
+
+            Some(Value::Bool(b.contains(&a)))
+        }
+        Function::NotIn(first_rule, second_rule) => {
+            let a = eval(input, output, first_rule)?.ok_or(Error::TypeError)?;
+            let b = eval(input, output, second_rule)?
+                .ok_or(Error::TypeError)?
+                .try_array()?;
+
+            Some(Value::Bool(!b.contains(&a)))
+        }
+        Function::Between(first_rule, second_rule, third_rule) => {
+            let first_eval = first_rule.eval(input, output)?.ok_or(Error::TypeError)?;
+            let second_eval = second_rule.eval(input, output)?.ok_or(Error::TypeError)?;
+            let third_eval = third_rule.eval(input, output)?.ok_or(Error::TypeError)?;
+
+            let value = BigNum::try_from(first_eval)?;
+            let start = BigNum::try_from(second_eval)?;
+            let end = BigNum::try_from(third_eval)?;
+
+            Some(Value::Bool(value.ge(&start) && value.le(&end)))
+        }
+        Function::At(first_rule, second_rule) => {
+            let first_eval = first_rule
+                .eval(input, output)?
+                .ok_or(Error::TypeError)?
+                .try_array()?;
+            let second_eval = second_rule
+                .eval(input, output)?
+                .ok_or(Error::TypeError)?
+                .try_number()?
+                .as_u64()
+                .ok_or(Error::TypeError)?;
+            let index = second_eval as usize;
+            let value = &first_eval[index];
+            Some(value.clone())
+        }
+        Function::Split(first_rule, second_rule) => {
+            let first_eval = first_rule
+                .eval(input, output)?
+                .ok_or(Error::TypeError)?
+                .try_string()?;
+            let second_eval = second_rule
+                .eval(input, output)?
+                .ok_or(Error::TypeError)?
+                .try_string()?;
+
+            let after_split: Vec<&str> = first_eval.split(&second_eval).collect();
```

Comment:
Instead of collecting and then creating an iterator again, just map the values before the `collect()`

Reply:
This still hasn't been resolved. The reason is that will impact the performance (collecting twice will result in memory allocation twice)

```rust
let after_split = first_eval.split(&second_eval).map(Value::new_string).collect();
Some(Value::Array(after_split))
```